packages/dht/src/connection/websocket/WebsocketConnector.ts

```diff
@@ -183,33 +183,29 @@ export class WebsocketConnector {
         for (const reattempt of range(ENTRY_POINT_CONNECTION_ATTEMPTS)) {
             const entryPoint = sample(this.config.entrypoints)!
             try {
-                if (!this.websocketServer) {
-                    return noServerConnectivityResponse
+                if (!this.config.entrypoints || this.config.entrypoints.length === 0) {
+                    // return connectivity info given in config
+                    const preconfiguredConnectivityResponse: ConnectivityResponse = {
+                        host: this.host!,
+                        natType: NatType.OPEN_INTERNET,
+                        websocket: { host: this.host!, port: this.selectedPort!, tls: this.config.tlsCertificate !== undefined },
+                        // TODO: maybe do a DNS lookup here?
+                        ipAddress: ipv4ToNumber('127.0.0.1'),
+                        version: localVersion
+                    }
+                    return preconfiguredConnectivityResponse
                 } else {
-                    if (!this.config.entrypoints || this.config.entrypoints.length === 0) {
-                        // return connectivity info given in config
-                        const preconfiguredConnectivityResponse: ConnectivityResponse = {
-                            host: this.host!,
-                            natType: NatType.OPEN_INTERNET,
-                            websocket: { host: this.host!, port: this.selectedPort!, tls: this.config.tlsCertificate !== undefined },
-                            // TODO: maybe do a DNS lookup here?
-                            ipAddress: ipv4ToNumber('127.0.0.1'),
-                            version: localVersion
-                        }
-                        return preconfiguredConnectivityResponse
+                    // Do real connectivity checking
+                    const connectivityRequest = {
+                        port: this.selectedPort ?? 0,
```

Comment:
Would it make sense to annotate `port` as optional field?

Reply:
I would prefer to avoid using optional fields (perhaps completely remove them in the future) since they are not standard protobuf. We use a custom flag of the `ts-protobuf` library to use them.

Reply:
The `optional` annotation is a good way to communicate whether we consider some field to be always included in the payload of valid messages or not. We can then build the entity validation based on that high level logic of field presence, and have correct TypeScript there too.

Do we currently have an accessor which we can use to check if an object contains a value for a field? That would very straightforward way to check if port was given or not.

Without optionality we need this kind of magic values. It is too easy to forget to handle the magic value case when we process the message.

Reply:
Seems that optional is supported since protobuf 3.15... Could change this to optional in the future.

Reply:
If we don't do that in this PR, could just introduce a constant for the 0 value. E.g. const `DISABLE_CONNECTIVITY_PROBE = 0`